### PR TITLE
Update FaultTolerance interfaces

### LINKED ISSUE #3515: FT: Add javadoc to ExecutorBuilder
ExecutorBuilder is a shared interface, it should definitely have javadoc, especially at the class level, and at the moment it has none.

### PR DESCRIPTION
**Pull request by Azquelt**
* Remove unused type parameter from `ExecutorBuilder`
* Add javadoc to `ExecutorBuilder` and `Executor`

Recommend reviewing commits separately

Fixes #3515

### CODE REVIEW

**Comment by Azquelt** [comment]
#build

File: dev/com.ibm.ws.microprofile.faulttolerance/src/com/ibm/ws/microprofile/faulttolerance/impl/ExecutorBuilderImpl.java [hunk 1]
```diff
@@ -27,7 +27,7 @@
 import com.ibm.ws.threading.PolicyExecutorProvider;
 import com.ibm.wsspi.threadcontext.WSContextService;
 
-public class ExecutorBuilderImpl<T, R> implements ExecutorBuilder<T, R> {
+public class ExecutorBuilderImpl<R> implements ExecutorBuilder<R> {
```
Comment (tevans78) [inline dev/com.ibm.ws.microprofile.faulttolerance/src/com/ibm/ws/microprofile/faulttolerance/impl/ExecutorBuilderImpl.java]: What is `R`?

Comment (Azquelt) [reply dev/com.ibm.ws.microprofile.faulttolerance/src/com/ibm/ws/microprofile/faulttolerance/impl/ExecutorBuilderImpl.java]: From the interface:
`* @param <R> the type of the result returned by {@code Executor}s created by this builder`

Comment (tevans78) [reply dev/com.ibm.ws.microprofile.faulttolerance/src/com/ibm/ws/microprofile/faulttolerance/impl/ExecutorBuilderImpl.java]: ok

**Comment by Azquelt** [comment]
Failures in the build do not appear to be caused by this change.

**Pull request merged**